topo_processor/metadata/collection_factory.py

```diff
@@ -1,28 +1,35 @@
+import asyncio
 import os
 
 from linz_logger import get_log
 
-from topo_processor.metadata.loaders import repo
+from topo_processor.metadata.loaders import loaders_repo
+from topo_processor.metadata.validators import validators_repo
 from topo_processor.util.time import time_in_ms
 
 from .collection import Collection
 from .data_type import DataType
 from .item import Item
 
 
-def create_collection(path: str, data_type: DataType) -> Collection:
+async def create_collection(path: str, data_type: DataType) -> Collection:
     start_time = time_in_ms()
     collection = Collection("title", "description", "license", data_type)
-    create_item(collection, path)
-    get_log().debug("CreateCollection", title=collection.title, data_type=data_type, duration=time_in_ms() - start_time)
+    await create_item(collection, path)
+    get_log().debug("Collection Created", title=collection.title, data_type=data_type, duration=time_in_ms() - start_time)
     return collection
 
 
-def create_item(collection: Collection, path: str) -> None:
+async def create_item(collection: Collection, path: str) -> None:
     start_time = time_in_ms()
+    list_to_add_metadata = []
+    list_to_validate_metadata = []
     for f in os.listdir(path):
         file_path = os.path.join(path, f)
         item = Item(file_path, collection)
-        repo.add_metadata(item)
+        list_to_add_metadata.append(loaders_repo.add_metadata(item))
+        list_to_validate_metadata.append(validators_repo.check_validity(item))
```

Comment:
you cannot do this like this, this will run them all concurrently the metadata checks need to happen before the validator checks.

I would suggest a `load_item()` function that  
```
await loaders_repo.add_metadata(..)
await validators_repo.check_validity(..)
```

Reply:
but if I 
```
    await asyncio.gather(*list_to_add_metadata)
    print("**************************************************")
    await asyncio.gather(*list_to_validate_metadata)
```
The console shows:
```
[2020-12-03T07:44:54.660Z] DEBUG Metadata Added loader=loader.imagery.historic duration=0.451904296875 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_C8054_A_2.tif
[2020-12-03T07:44:54.665Z] DEBUG Metadata Added loader=loader.imagery.tiff duration=4.681640625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_C8054_A_2.tif
[2020-12-03T07:44:54.665Z] DEBUG Metadata Added loader=loader.checksum duration=0.258056640625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_C8054_A_2.tif
[2020-12-03T07:44:54.665Z] DEBUG Metadata Added loader=loader.imagery.historic duration=0.05029296875 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_49.tiff
[2020-12-03T07:44:54.667Z] DEBUG Metadata Added loader=loader.imagery.tiff duration=1.69970703125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_49.tiff
[2020-12-03T07:44:54.667Z] DEBUG Metadata Added loader=loader.checksum duration=0.175537109375 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_49.tiff
[2020-12-03T07:44:54.667Z] DEBUG Metadata Added loader=loader.imagery.historic duration=0.06689453125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_48.tiff
[2020-12-03T07:44:54.669Z] DEBUG Metadata Added loader=loader.imagery.tiff duration=1.584228515625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_48.tiff
[2020-12-03T07:44:54.669Z] DEBUG Metadata Added loader=loader.checksum duration=0.09814453125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_48.tiff
[2020-12-03T07:44:54.669Z] DEBUG Metadata Added loader=loader.imagery.historic duration=0.0556640625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_51.tiff
[2020-12-03T07:44:54.684Z] DEBUG Metadata Added loader=loader.imagery.tiff duration=14.75634765625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_51.tiff
[2020-12-03T07:44:54.684Z] DEBUG Metadata Added loader=loader.checksum duration=0.1513671875 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_51.tiff
[2020-12-03T07:44:54.685Z] DEBUG Metadata Added loader=loader.imagery.historic duration=0.046142578125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_50.tiff
[2020-12-03T07:44:54.686Z] DEBUG Metadata Added loader=loader.imagery.tiff duration=1.889892578125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_50.tiff
[2020-12-03T07:44:54.687Z] DEBUG Metadata Added loader=loader.checksum duration=0.1181640625 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_50.tiff
**************************************************
[2020-12-03T07:44:54.688Z] DEBUG Validity Checked validator=validator.imagery.tiff duration=1.50146484375 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_C8054_A_2.tif
[2020-12-03T07:44:54.690Z] DEBUG Validity Checked validator=validator.imagery.tiff duration=1.772216796875 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_49.tiff
[2020-12-03T07:44:54.692Z] DEBUG Validity Checked validator=validator.imagery.tiff duration=1.399658203125 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_48.tiff
[2020-12-03T07:44:54.694Z] DEBUG Validity Checked validator=validator.imagery.tiff duration=1.9990234375 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_51.tiff
[2020-12-03T07:44:54.696Z] DEBUG Validity Checked validator=validator.imagery.tiff duration=1.7421875 path=/home/bcheng/dev/topo-processor/test_data/tiffs/CROWN_399_E_50.tiff
[2020-12-03T07:44:54.696Z] DEBUG Items Created path=/home/bcheng/dev/topo-processor/test_data/tiffs duration=37.082763671875
[2020-12-03T07:44:54.696Z] DEBUG Collection Created title=title data_type=<DataType.ImageryHistoric: 'imagery.historic'> duration=37.174560546875
```